lib/util/connection-leasing.js

```diff
@@ -0,0 +1,26 @@
+const LeaseConnectionFromClient = (client) =>
+  async function withConnection(next) {
+    const connection = await client.acquireConnection();
+    try {
+      return await next(connection);
+    } finally {
+      await client.releaseConnection(connection);
+    }
+  };
+
+function ReuseConnection(connection) {
```

Comment:
PascalCase looks confusing here; this doesn't look like a class or class-like function. Would it be possible to clarify naming a bit, to make it more self-explanatory what functions in this file do?

Reply:
Ah -- I tend to use PascalCase for factory functions as well.  In this case: it's a function that returns a connection leasing function.

Reply:
If it's a factory function, then probably it should be a noun and not a verb.

Reply:
However, since this is a factory function for making functions... Maybe naming should explain that fact somehow?

Reply:
Makes sense.  I'll take care of that in a little bit.

I'm also gonna post a "thinking out loud" comment in a little bit to clarify some of the other things that seem to relate to all of this.  (Ex: connection pooling, etc)